src/Database/Traits/Validation.php

```diff
@@ -435,31 +436,42 @@ protected function processRuleFieldNames($rules)
      */
     protected function processValidationUniqueRule($definition, $fieldName)
     {
+        $connection = '';
         list(
             $table,
             $column,
-            $key,
-            $keyName,
+            $ignoreValue,
+            $ignoreColumn,
             $whereColumn,
             $whereValue
         ) = array_pad(explode(',', $definition, 6), 6, null);
 
-        $table = 'unique:' . $this->getConnectionName()  . '.' . $this->getTable();
-        $column = $column ?: $fieldName;
-        $key = $keyName ? $this->$keyName : $this->getKey();
-        if (is_null($key)) {
-            $key = 'NULL';
+        // Remove unique or unique: from the table name
+        $table = ltrim(ltrim($table, 'unique'), ':');
+
+        // Support table, connection.table, and null value
+        if (Str::contains($table, '.')) {
+            $connectionDetails = explode('.', $table);
+            $connection = $connectionDetails[0];
+            $table = $connectionDetails[1];
+        } elseif (empty($table)) {
+            $table = $this->getTable();
+        }
+        if (empty($connection)) {
+            $connection = $this->getConnectionName();
         }
 
-        $keyName = $keyName ?: $this->getKeyName();
+        $column = $column ?: $fieldName;
+        $ignoreColumn = $ignoreColumn ?: $this->getKeyName();
+        $ignoreValue = ($ignoreValue && $ignoreValue !== 'NULL') ? $ignoreValue : $this->{$ignoreColumn};
+        if (is_null($ignoreValue)) {
+            $ignoreValue = 'NULL';
+        }
 
-        $params = [$table, $column, $key, $keyName];
+        $params = ["unique:{$connection}.{$table}", $column, $ignoreValue, $ignoreColumn];
 
-        if ($whereColumn) {
+        if ($whereColumn && $whereValue) {
             $params[] = $whereColumn;
-        }
-
-        if ($whereValue) {
             $params[] = $whereValue;
         }
```

Comment:
@LukeTowers would this break conditions that don't have a value - not sure if it's possible, but I wonder if this method previously supported things like "ISNULL(deleted_at)", which technically does not need a value to be used as a condition.

Reply:
@bennothommo good catch, looks like that is supported in the Unique rule object at least: https://github.com/laravel/framework/blob/8.x/src/Illuminate/Validation/Rules/DatabaseRule.php#L93-L95. Mind adding a test for it and pushing the fix?